lib/api/CDataFrameAnalysisRunner.cc

```diff
@@ -23,7 +24,7 @@ std::size_t memoryLimitWithSafetyMargin(const CDataFrameAnalysisSpecification& s
     return static_cast<std::size_t>(0.9 * static_cast<double>(spec.memoryLimit()) + 0.5);
 }
 
-const double MAXIMUM_FRACTIONAL_PROGRESS{1024.0};
+const std::size_t MAXIMUM_FRACTIONAL_PROGRESS{1ull << ((sizeof(std::size_t) - 2) * 8)};
```

Comment:
Instead of `1ull` I would use `std::size_t(1)`.  On some platforms `std::size_t` is not `unsigned long long`, nor even the same size as it.

Reply:
My thinking was the key requirement was that whatever type the shift operation didn't overflow, but I agree that is probably the cleanest.

Reply:
[a0c6874](https://github.com/elastic/ml-cpp/pull/368/commits/a0c6874634ced3ca48446c87852028de81725b22)